x-pack/plugins/apm/server/lib/connections/get_connection_metrics/get_metrics.ts

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sum } from 'lodash';
+import objectHash from 'object-hash';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/api/types';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import { getOffsetInMs } from '../../../../common/utils/get_offset_in_ms';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../common/environment_filter_values';
+import { asMutableArray } from '../../../../common/utils/as_mutable_array';
+import {
+  AGENT_NAME,
+  EVENT_OUTCOME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  SPAN_DESTINATION_SERVICE_RESOURCE,
+  SPAN_DESTINATION_SERVICE_RESPONSE_TIME_COUNT,
+  SPAN_DESTINATION_SERVICE_RESPONSE_TIME_SUM,
+  SPAN_SUBTYPE,
+  SPAN_TYPE,
+} from '../../../../common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import { rangeQuery } from '../../../../../observability/server';
+import { getBucketSize } from '../../helpers/get_bucket_size';
+import { EventOutcome } from '../../../../common/event_outcome';
+import { Setup } from '../../helpers/setup_request';
+import { NodeType } from '../../../../common/connections';
+
+export const getMetrics = async ({
+  setup,
+  start,
+  end,
+  filter,
+  numBuckets,
+  offset,
+}: {
+  setup: Setup;
+  start: number;
+  end: number;
+  filter: QueryDslQueryContainer[];
+  numBuckets: number;
+  offset?: string;
+}) => {
+  const { apmEventClient } = setup;
+
+  const offsetInMs = getOffsetInMs(start, offset);
+  const startWithOffset = start - offsetInMs;
+  const endWithOffset = end - offsetInMs;
+
+  const response = await apmEventClient.search(
+    'get_service_destination_metrics',
+    {
+      apm: {
+        events: [ProcessorEvent.metric],
+      },
+      body: {
+        track_total_hits: true,
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...filter,
+              {
+                exists: {
+                  field: SPAN_DESTINATION_SERVICE_RESPONSE_TIME_COUNT,
+                },
+              },
+              ...rangeQuery(startWithOffset, endWithOffset),
+              {
+                bool: {
+                  must_not: [{ term: { [AGENT_NAME]: 'rum-js' } }],
```

Comment:
a comment here too?

Reply:
Probably want to use [RUM_AGENT_NAMES](https://github.com/elastic/kibana/blob/f067ac866005e0e90138fde9fea69450b24203be/x-pack/plugins/apm/common/agent_name.ts#L55) to include legacy and otel names.